apache/hive/weak_credentials/README.md

```diff
@@ -0,0 +1,107 @@
+# setup 
+## vulnerable instance
+### none authorization
+
+1.run the hive instance with default mode, empty name and password
+
+```bash
+$ docker run -d -p 10000:10000 -p 10002:10002 --env SERVICE_NAME=hiveserver2 --name hive4 apache/hive:4.0.0
+```
+
+2.ensure status of hive with open `http(s)://hive_addr:10002`
+
+3.connect hive without authorization for testing and it works
+
+```shell
+$ docker exec -it hive4 beeline -u 'jdbc:hive2://localhost:10000/'
+```
+### ldap authorization with username and password
+1.install ldap
+
+```shell
+$ docker-compose up -d
+$ docker cp base.ldif my-openldap-container:/base.ldif
+$ docker exec -it my-openldap-container ldapadd -x -D "cn=admin,dc=example,dc=com" -w adminpassword -f /base.ldif
+```
+2.ensure that the normal account is set up properly
+
+open `https://localhost:6443`, login DN is `cn=admin,dc=example,dc=com` and password is `adminpassword` from `docker-compose.yml`
+
+3.edit `hive-site.xml` and replace `ldap://192.168.64.1:389` to `ldap://your_ldap_ip:389`(`localhost` or `127.0.0.1`
+is not useful, I think using the host address is better)
```

Comment:
I'm not understanding this part. Why do i need to change the IP address in the configuration and what IP is supposed to be put in there? I'm using the public IP of the VM where i'm deploying the container, but it seems not to work later on.

Reply:
Hi @W0ngL1. I can login to phpldapadmin correctly

Reply:
Perfect. By using the IP address of the docker bridge interface everything works fine. I think you should definitely add this in the testbed explanation.